src/azul/plugins/metadata/hca/transform.py

```diff
@@ -138,32 +155,28 @@ def to_index(self, value: Optional[JSON]) -> str:
         Traceback (most recent call last):
         ...
         azul.RequirementError: The `unit` entry must be a string
-
         """
         if value is None:
             return NullableString.null_string
         else:
+            msg = 'A dictionary with entries for `value` and `unit` is required'
+            require(len(value) == 2, msg)
             try:
                 value, unit = value['value'], value['unit']
             except KeyError:
-                raise RequirementError(
-                    'A dictionary with entries for `value` and `unit` is required'
-                )
+                reject(True, msg)
             else:
                 reject(value == '', 'The `value` entry must not be empty')
                 reject(unit == '', 'The `unit` entry must not be empty')
-        if value is None:
-            raise RequirementError(
-                'The `value` entry must not be null'
-            )
-        elif unit is None:
-            return value
-        else:
-            require(type(value) is str, 'The `value` entry must be a string')
-            require(type(unit) is str, 'The `unit` entry must be a string')
-            reject(' ' in value, 'The `value` entry must not contain space characters')
-            reject(' ' in unit, 'The `unit` entry must not contain space characters')
-            return f'{value} {unit}'
+                reject(value is None, 'The `value` entry must not be null')
+                if unit is None:
+                    return value
```

Comment:
Don't we need to apply more checks on `value` before returning it? Why don't the doctests cover this?

Reply:
Completely, that slipped through the cracks, my bad.